src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -91,14 +109,26 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
         if (!this._isMounted) return;
         this.setState({
           selectedOption: [{ id: optionId, label: selectedDataSource.title }],
+          defaultDataSource,
         });
         if (this.props.onSelectedDataSources) {
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
+        this.setState({
+          selectedOption: [],
+          defaultDataSource,
+        });
+        if (this.props.onSelectedDataSources) {
+          this.props.onSelectedDataSources([]);
+        }
         handleDataSourceFetchError(this.onError.bind(this), this.props.notifications!);
```

Comment:
can we see if we can remove line 118-124 and use the handleDataSourceFetchError instead?